python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -1103,7 +1105,10 @@ def ngroup(self, ascending=True):
         """
         index = self.grouping.keys.unique().sort_values()
         num_groups = len(index)
-        _, has_null_group = bitmask_or([*index._columns])
+        with acquire_spill_lock():
+            _, has_null_group = plc.null_mask.bitmask_or(
+                [col.to_pylibcudf(mode="read") for col in index._columns]
+            )
```

Comment:
FWIW, this can be computed without a kernel, because the columns know their null count, so this is equivalent to:

```
has_null_group = any(col.has_nulls() for col in index._columns)
```